### DISCUSSION TITLE
is the component class available for laravel 6?

### DISCUSSION THREAD

**Discussion by videni** [laravel/framework#41135] [General]
anyone integrate component class for laravel 6?  I got legacy project which can't upgrade to laravel >= 7.

**Comment by gguney** [comment]
As my understanding you are trying to use component tag in Laravel's blade files? [here](https://laravel.com/docs/6.x/blade#components-and-slots) you can see how to use component directive. If I misunderstood something please don't hesitate to ask.

**Comment by videni** [comment]
@gguney , thanks for quick response, I mean this 

```
amespace App\View\Components;
 
use Illuminate\View\Component;
 
class Alert extends Component
{
    /**
     * The alert type.
     *
     * @var string
     */
    public $type;
 
    /**
     * The alert message.
     *
     * @var string
     */
    public $message;
 
    /**
     * Create the component instance.
     *
     * @param  string  $type
     * @param  string  $message
     * @return void
     */
    public function __construct($type, $message)
    {
        $this->type = $type;
        $this->message = $message;
    }
 
    /**
     * Get the view / contents that represent the component.
     *
     * @return \Illuminate\View\View|\Closure|string
     */
    public function render()
    {
        return view('components.alert');
    }
}
```

It works as view model for a component.

**Reply by gguney** [reply]
Ah, I see, We do not have it in Laravel 7 but again you can follow https://laravel.com/docs/6.x/blade#components-and-slots this link and use blade components, it will serve you the same way.
